tests/Integration/Momento.Sdk.Tests/DictionaryTest.cs

```diff
@@ -0,0 +1,1123 @@
+    {
+        var dictionaryName = Utils.NewGuidString();
+        var fieldName = Utils.NewGuidString();
+
+        var setResponse = await client.DictionarySetFieldAsync(cacheName, dictionaryName, fieldName, "abcxyz");
+        Assert.True(setResponse is CacheDictionarySetFieldResponse.Success, $"Unexpected response: {setResponse}");
+
+        var dictionaryIncrementResponse = await client.DictionaryIncrementAsync(cacheName, dictionaryName, fieldName);
+        Assert.True(dictionaryIncrementResponse is CacheDictionaryIncrementResponse.Error, $"Unexpected response: {dictionaryIncrementResponse}");
+        Assert.Equal(MomentoErrorCode.FAILED_PRECONDITION_ERROR, ((CacheDictionaryIncrementResponse.Error)dictionaryIncrementResponse).ErrorCode);
+    }
+
+    [Theory]
+    [InlineData(null, "my-dictionary", "my-field")]
+    [InlineData("cache", null, "my-field")]
+    [InlineData("cache", "my-dictionary", null)]
+    public async Task DictionaryGetFieldAsync_NullChecksFieldIsString_IsError(string cacheName, string dictionaryName, string field)
+    {
+        CacheDictionaryGetFieldResponse response = await client.DictionaryGetFieldAsync(cacheName, dictionaryName, field);
+        Assert.True(response is CacheDictionaryGetFieldResponse.Error, $"Unexpected response: {response}");
+        var errResponse = (CacheDictionaryGetFieldResponse.Error)response;
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((CacheDictionaryGetFieldResponse.Error)response).ErrorCode);
+        Assert.Equal(field, ((CacheDictionaryGetFieldResponse.Error)response).FieldString);
+    }
+
+    [Theory]
+    [InlineData(null, "my-dictionary", "my-field", "my-value")]
+    [InlineData("cache", null, "my-field", "my-value")]
+    [InlineData("cache", "my-dictionary", null, "my-value")]
+    [InlineData("cache", "my-dictionary", "my-field", null)]
+    public async Task DictionarySetFieldAsync_NullChecksFieldIsStringValueIsString_IsError(string cacheName, string dictionaryName, string field, string value)
+    {
+        CacheDictionarySetFieldResponse response = await client.DictionarySetFieldAsync(cacheName, dictionaryName, field, value);
+        Assert.True(response is CacheDictionarySetFieldResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((CacheDictionarySetFieldResponse.Error)response).ErrorCode);
+    }
+
+    [Fact]
+    public async Task DictionaryGetFieldAsync_FieldIsString_DictionaryIsMissing()
+    {
+        var dictionaryName = Utils.NewGuidString();
+        var field = Utils.NewGuidString();
+        CacheDictionaryGetFieldResponse response = await client.DictionaryGetFieldAsync(cacheName, dictionaryName, field);
+        Assert.True(response is CacheDictionaryGetFieldResponse.Miss, $"Unexpected response: {response}");
+        var missResponse = (CacheDictionaryGetFieldResponse.Miss)response;
+        Assert.Equal(field, missResponse.FieldString);
```

Comment:
New checks related to `FieldString` and `FieldByteArray` are added.
This is to check if the returned value matches `field` provided by users